packages/x-adapter/src/http-clients/utils.ts

```diff
@@ -13,7 +13,7 @@ import { RequestError } from './errors/request-error';
  */
 export function toJson(response: Response): Promise<any> {
   if (response.ok) {
-    return response.json();
+    return response.text().then(text => text && JSON.parse(text));
```

Comment:
Maybe I'm thinking too much here but what if the response is not a valid JSON? Should we consider such scenario hand and have handler for it?

Reply:
You are right, but we haven't developed a proper error handling yet, so I don't know what you would like to do here, if just raising an error with a friendlier message or what. I'm open to it though. 

This code should be equivalent with the previous one but with support for empty strings. What I'm thinking right now is if it makes sense to return a string when we expect a JSON object, or should I return an empty object instead? This wouldn't have happened if this function wasn't using `any` 🤣. Truth is that it returns something `unknown` here, and as we don't know the type the only safe way would be to cast it whenever this `toJson` function is used.

Reply:
I was thinking on catching the error, logging a descriptive message and return `null` and this way we prevent the uncaught exception. However, you are right and this is not within the scope of this PR and just something to think about in the future.